src/autoscaler/api/cmd/api/main.go

```diff
@@ -80,10 +81,10 @@ func main() {
 		os.Exit(1)
 	}
 
-	pm := cred_helper.PluginManager{}
+	pm := plugin.Manager{}
 	defer pm.Kill()
 
-	credentials, err := pm.LoadCredentialPlugin(conf.DB, conf.Logging, conf.CredHelperPluginPath)
+	credentials, err := pm.LoadCredentialsImplementation(conf.DB, conf.Logging, conf.CredHelperPlugin, conf.StoredProcedureConfig)
```

Comment:
hmmm it would seem to be much  simpler if we had a simple if ...
```
if conf.StoredProcedureConfig {
 credentials, err = NewStoredProc()
} else if conf.CredHelperPlugin {
 credentials, err = NewCredHelper()
} else {
  err = ERROR 
}
```
where `NewCredProcedure` is a the constructor for stored procs  
    and `NewCredPlugin` is the plugin constructor (which also does the `LoadCredentialsImplementation` internally)

There is no reall need for the plugin manager in the case of stored procedures.